Makefile

```diff
@@ -76,19 +73,20 @@ docs: ## generate Sphinx HTML documentation, including API docs
 
 # Define PIP_COMPILE_OPTS=-v to get more information during make upgrade.
 PIP_COMPILE = pip-compile --upgrade $(PIP_COMPILE_OPTS)
+CONSTRAINTS = requirements/edx-platform-constraints.txt
 
-upgrade: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
+upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
+upgrade:  ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
+	echo "### DON'T edit this file, it's copied from edx-platform. See make upgrade" > $(CONSTRAINTS)
+	curl -fsSL https://raw.githubusercontent.com/edx/edx-platform/master/requirements/edx/base.txt | grep -v '^-e' >> $(CONSTRAINTS)
 	pip install -q pip-tools
-	$(PIP_COMPILE) -o requirements/dev.txt requirements/base.in requirements/test-$(TARGET_PLATFORM).in requirements/dev.in requirements/quality.in
-	$(PIP_COMPILE) -o requirements/doc.txt requirements/base.in requirements/test-$(TARGET_PLATFORM).in requirements/doc.in
-	$(PIP_COMPILE) -o requirements/quality.txt requirements/base.in requirements/test-$(TARGET_PLATFORM).in requirements/dev.in requirements/quality.in requirements/doc.in requirements/test.in
+	$(PIP_COMPILE) -o requirements/base.txt requirements/base.in
+	$(PIP_COMPILE) -o requirements/dev.txt requirements/dev.in requirements/quality.in
```

Comment:
Do we still need a distinction between `dev.txt` and `quality.txt` here?  It looks like the former is a subset of the latter, and I'd think that developers would want to be able to run any of the quality checks in their dev environment.